src/Bedrock.Framework/Protocols/WebSockets/WebSocketFrameReader.cs

```diff
@@ -21,17 +28,59 @@ public struct WebSocketFrameReader : IMessageReader<WebSocketReadFrame>
         /// <returns>True if parsed successfully, false otherwise.</returns>
         public bool TryParseMessage(in ReadOnlySequence<byte> input, ref SequencePosition consumed, ref SequencePosition examined, out WebSocketReadFrame message)
         {
-            var reader = new SequenceReader<byte>(input);
-
             //We need to at least be able to read the start of frame header
             if (input.Length < 2)
             {
                 message = default;
                 return false;
             }
 
-            reader.TryRead(out var finOpcodeByte);
-            reader.TryRead(out var maskLengthByte);
+            if (input.IsSingleSegment || input.FirstSpan.Length >= 14)
+            {
+                if (TryParseSpan(input.FirstSpan, input.Length, out var bytesRead, out message))
+                {
+                    consumed = input.GetPosition(bytesRead);
+                    examined = consumed;
+
+                    return true;
+                }
+
+                return false;
+
+            }
+            else
+            {
+                Span<byte> tempSpan = stackalloc byte[14];
+
+                var bytesToCopy = Math.Min(input.Length, tempSpan.Length);
+                input.Slice(0, bytesToCopy).CopyTo(tempSpan);
```

Comment:
Wouldn't this fail with `ArgumentException` that bytesToCopy is outside the range if input has length of less than `14` bytes?
I think there should be a check somewhere right?